Urban-Fun/View Controllers/SearchViewController.m

```diff
@@ -6,26 +6,225 @@
+    [attributesForNormal setObject:font forKey:NSFontAttributeName];
+    [self.searchControl setTitleTextAttributes:attributesForNormal
+                                    forState:UIControlStateNormal];
+    self.manager = [[CLLocationManager alloc] init];
+    self.manager.delegate = self;
+    self.manager.desiredAccuracy = kCLLocationAccuracyHundredMeters; //battery
+    [self.manager requestWhenInUseAuthorization];
+    if (CLLocationManager.locationServicesEnabled){
+        [self.manager startUpdatingLocation];
+    }
+}
+
+- (void) locationManager:(CLLocationManager *)manager didUpdateLocations:(NSArray<CLLocation *> *)locations{
+    if (locations.firstObject){
+        [manager stopUpdatingLocation];
+        self.currentUserLocation = locations.firstObject;
+    }
+}
+
+- (IBAction)changedSearchType:(id)sender {
+    NSString *searchText = self.searchBar.text;
+    if (self.searchControl.selectedSegmentIndex == 0){
+        //removing white space
+        searchText = [searchText stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
+        PFQuery *activityQuery = [Activity query];
+        [activityQuery whereKey:@"title" matchesRegex:searchText modifiers:@"i"];
+        [activityQuery orderByDescending:@"createdAt"];
+        [activityQuery includeKey:@"host"];
+        self.activitiesArray = [[NSMutableArray alloc] init];
+        [activityQuery findObjectsInBackgroundWithBlock:^(NSArray<Activity *>*activities , NSError *error) {
+            if (error) {
+                
+            }
+            for (Activity *activity in activities) {
+                if (![self.activitiesArray containsObject:activity]){
+                    [self.activitiesArray addObject:activity];
+                }
+            }
+            [self.tableView reloadData];
+
+        }];
+    }
+    if (self.searchControl.selectedSegmentIndex == 1){
+        //removing white space
+        searchText = [searchText stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
+        PFQuery *userQuery = [User query];
+        [userQuery whereKey:@"username" matchesRegex:searchText modifiers:@"i"];
+        [userQuery includeKey:@"host"];
+        self.usersArray = [[NSMutableArray alloc] init];
+        [userQuery findObjectsInBackgroundWithBlock:^(NSArray<User *>*users , NSError *error) {
+            if (error) {
+                
+            }
+            for (User *user in users) {
+                if (![self.usersArray containsObject:user]){
+                    [self.usersArray addObject:user];
+                }
+            }
+            [self.tableView reloadData];
+
+        }];
+    }
+}
+
+- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
+    if (self.searchControl.selectedSegmentIndex == 0){
+        //removing white space
```

Comment:
🚧 Remember to reuse code: Instead of having the same code here, we should put this code into a helper class or at least in some method to replace the call, we don't want to have the same code in different places because the maintenance cost long term is high